### DISCUSSION TITLE
Adding a node to the editor is annoying sometimes

### DISCUSSION THREAD

**Discussion by PabloDevGit** [godotengine/godot-proposals#4056] [Editor]
Straight to the point.
You add an enemy to your level map and ohh, you forgot you have an "item" node selected, so the enemy node will become a child of the "item" node, in this gif you can see. In the video I do it conscientiously, so it serves as an example.

https://user-images.githubusercontent.com/19652075/154941824-f885a014-6341-44e7-996d-69e668415090.mp4

If you don't realize that, then you will experience strange running errors until you realize that the node is being improperly childed by another node.
Add an option or some way to avoid this. If the user adds a node to the editor, it is assumed that they are adding it to the root node, right?.
An option could be added to the editor that states that when adding a node to the editor, the node should be added as a child of the root node, not the currently selected node.

**Comment by Calinou** [comment]
I think it would make sense to add a node as a *sibling* (rather than a child) if you hold <kbd>Shift</kbd> while pressing <kbd>Enter</kbd> (or clicking the **Create** button) in the Create New Node dialog. It would be difficult to discover though.

**Reply by SilencedPerson** [reply]
Also dragging from file system into the scene tree would be mighty helpful in mitigating this issue. simply placing the object at 0,0 or 0,0,0 would be enough in most cases.

**Comment by azur-wolve** [comment]
The workflow of: by default add to root, unless you specify the specific parent on the scene (via a creation dialogue, hotkey or whatever) looks like more intuitive;
it assumes root_node level by default and if you want it in an specific hierarchy_level you specify it explicitly.

It makes sense.

**Comment by andyp123** [comment]
Yes, this drives me crazy sometimes - especially in 3d, as I often add things accidentally as a sibling and they inherit the scale of the parent, which is not always 1,1,1 if it's part of the environment, and particularly test environments, where I'm more likely to scale or rotate a shape just to quickly build level geo. If something gets scaled wrong and you move it out of the parent object, the wrong scale is retained, so you need to just undo and do it again.

copy & paste seems to work in the same way as dropping things from the file tree, whereas duplicate (ctrl+d) will make a copy at the same level as the original node/scene (sibling), which is usually what I want in that case.

I totally agree that dropping things into the scene view should assume root node, and dropping onto the scene tree should allow you to be explicit about the parenting. Copy and paste should either follow this same rule, or just be made the same as duplicate.

**Comment by Mickeon** [comment]
Maybe some other line of text following the mouse **warning** about what you're going to do as you're dragging the new Node could be a quick and dirty way to mitigate this? 

Something like "*Adding as a child of ____...*"

**Reply by azur-wolve** [reply]
It could mitigate, but as you use it more often, eventually can start to don't read it and click "ok" fast.
The other approach of adding to scene_root by default looks like more "workflow-solid".

**Reply by Mickeon** [reply]
So perhaps this other line of text could hint at the alternative mode of operation.

("*Hold CTRL to add as a child of ____...*")
